src/components/menu/menu-item/menu-item.component.js

```diff
@@ -129,6 +132,7 @@ const MenuItem = ({
       menuType={menuContext.menuType}
       inSubmenu={submenuContext.handleKeyDown !== undefined}
       display="inline-block"
+      title={maxWidth && typeof children === "string" ? children : ""}
```

Comment:
maybe you should create a function?
some like:
`const getTitle = title => maxWidth && typeof title === "string" ? title : ""`